presto-hive/src/main/java/com/facebook/presto/hive/util/BackgroundBatchCacheLoader.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.util;
+
+import com.google.common.cache.CacheLoader;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import io.airlift.log.Logger;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public abstract class BackgroundBatchCacheLoader<K, V>
+        extends CacheLoader<K, V>
+{
+    private static final Logger logger = Logger.get(BackgroundBatchCacheLoader.class);
+    private ConcurrentMap<K, SettableFuture<V>> futureMap = Maps.newConcurrentMap();
+    private final int batchLoadSize;
+
+    private final ScheduledExecutorService scheduledExecutorService =
+            Executors.newScheduledThreadPool(5, daemonThreadsNamed("batch-cache-loader-%d"));
+
+    protected BackgroundBatchCacheLoader(final long batchLoadIntervalMillis, final int batchLoadSize)
+    {
+        checkArgument(batchLoadIntervalMillis > 0, "batchLoadIntervalMillis must be a positive integer.");
+        checkArgument(batchLoadSize > 0, "batchLoadSize must be a positive integer.");
+        this.batchLoadSize = batchLoadSize;
+
+        Runnable timeThresholdReachedRunnable = createRunnable(null);
+        scheduledExecutorService.scheduleWithFixedDelay(timeThresholdReachedRunnable, batchLoadIntervalMillis, batchLoadIntervalMillis, MILLISECONDS);
```

Comment:
Also, pass in the ExecutorService to the constructor so that different BackgroundBatchCacheLoader instances can share the same ExecutorService